packages/client/karma-setup.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * This setup script is executed indirectly by Karma via the `browser-test-runner`
+ * package. See karma-config.js in that package for more details.
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const { toThrowStreamrError } = require('./test/test-utils/customMatchers')
+
+if (toThrowStreamrError !== undefined) {
```

Comment:
In which situation `toThrowStreamrError` can be `undefined`?

Reply:
There's something weird about the Karma runner here that requires this undefined check.

Tests that use `toThrowStreamrError` definitely have it available as they pass without problem. If I don't do this setup I will see that those tests would error. So we are definitely extending jest matches with something that is defined.

_However_ at the very end of the test suite, Karma will report that we are trying to extend an `undefined` matcher. My guess is that Karma is for some reason running this setup script many times and perhaps at the very end the  `./test/test-utils/customMatchers` is somehow unloaded and thus the variable becomes `undefined`. 

But it's a bit of a mystery and imo not worth delving more time into.